sympl/_components/netcdf.py

```diff
@@ -50,11 +50,30 @@ def __init__(
                 This can result in many file open/close operations.
                 Default is to write only when the write() method is
                 called directly.
+            aliases : dict
+                A dictionary used to rename variables to their aliases.
+                keys = variable name; values = alias name
+                The function sympl.get_component_aliases() can be used to
+                get variable aliases from a list of Components
             """
             self._cached_state_dict = {}
             self._filename = filename
             self._time_units = time_units
             self._write_on_store = write_on_store
+            if aliases is None:
+                self._aliases = {}
+            else:
+                self._aliases = aliases
+            popkeys = []
```

Comment:
This popping keys section should either be removed, or fixed and made to raise an exception instead of silently popping keys. If you want to fix it, these are some things that you should do:

- If a bad thing happens, `raise ValueError("<put description here>")` or `raise TypeError("<put description here>") instead of popping. `TypeError` should be used for non-string stuff, `ValueError` should be used for empty string names.
- Check for a string by putting `from six import string_types at the top of the file, and then using `isinstance(var, string_types)`. This is important for 2 and 3 compatibility when using byte strings vs char strings.
- Don't raise an exception immediately if a target of an alias is the empty string. Instead, raise an exception if a variable in the state is going to be aliased to the empty string. For example, a user might for some reason want to remove "air_" from all variable names by aliasing "air_" to the empty string. The exception you raise should describe to the user that the problem is due to an alias (in other words, it should give more information than the one xarray would give anyways when you try to store the variable with an empty string name).